submissions/HannaSyn/oop-exercise/index.js

```diff
@@ -0,0 +1,90 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+   constructor ( name, gender, friends, say, biologicalSpecies ) {
+      this.name = name;
+      this.biologicalSpecies = biologicalSpecies;
+      this.gender = gender;
+      this.friends = friends;
+      this.say = say
+   }
+   toString() {
+      return `${this.say} I'm ${this.biologicalSpecies}! My name is ${this.name}, my gender is ${this.gender}, and I have friends: ${this.friends}.`;
+   }
+}
+
+class Human extends Inhabitant {
+   constructor( name, gender, friends, say, legs, hands ) {
+      super( name, gender, friends, say, 'human');
+      this.legs = 2 || legs;
+      this.hands = 2 || hands;
+   } 
+   toString() {
+      return super.toString() + ` I have ${this.legs} legs and ${this.hands} hands.`
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor( name, gender, friends, say, biologicalSpecies, paws) {
+      super( name, gender, friends, say, biologicalSpecies )
+      this.paws = 4 || paws;
```

Comment:
This is a valid approach.
However do not forget that `constructor` is a function so it allows for default values `constructor(a, b, c = 5) { ... }`
Well, the latter approach doesn't work well when you have dafaults for the arguments in the middle. This means that all args with defaults should be in the end of args list.
You may also combine both approaches.
Just FYI

Reply:
Should I fix this?